### PR TITLE
Add TestEtcdStoragePath integration test

### PR DESCRIPTION
**Pull request by enj**
**What this PR does / why we need it**:

`TestEtcdStoragePath` prevents the accidental movement of objects stored in etcd.  It creates a stub of each object and then checks the expected location in etcd.  Inadvertent `GroupVersionKind` changes are prevented.

cc @liggitt @soltysh for review.

cc @ncdc because I know you love this 😄 

Signed-off-by: Monis Khan <mkhan@redhat.com>

**Release note**:

```
NONE
```

### CODE REVIEW

**Comment by liggitt** [comment]
@k8s-bot ok to test

File: test/integration/etcd/etcd_storage_path_test.go [hunk 1]
```diff
@@ -0,0 +1,933 @@
...
+
+	kubeClientConfigValue := atomic.Value{}
+	storageConfigValue := atomic.Value{}
+
+	go func() {
+		for {
+			kubeAPIServerOptions := options.NewServerRunOptions()
+			kubeAPIServerOptions.SecureServing.BindAddress = net.ParseIP("127.0.0.1")
+			kubeAPIServerOptions.Etcd.StorageConfig.ServerList = []string{framework.GetEtcdURLFromEnv()}
+			kubeAPIServerOptions.Etcd.DefaultStorageMediaType = runtime.ContentTypeJSON // TODO use protobuf?
```
Comment (soltysh) [inline test/integration/etcd/etcd_storage_path_test.go]: We force protobuf in [api-server](https://github.com/kubernetes/kubernetes/blob/a0418d01ac6a332f007a4a0ca639f4acede30edd/cmd/kube-apiserver/app/options/options.go#L112) so I'd say it should be protobuf here as well.

File: test/integration/etcd/etcd_storage_path_test.go [hunk 2]
```diff
@@ -0,0 +1,933 @@
...
+		}
+
+		kind := gvk.Kind
+		pkgPath := apiType.PkgPath()
+
+		mapping, err := mapper.RESTMapping(gvk.GroupKind(), gvk.Version)
+		if err != nil {
+			kindSeen.Insert(kind)
+			if kindWhiteList.Has(kind) {
+				// t.Logf("skipping test for %s from %s because its GVK %s is whitelisted and has no mapping", kind, pkgPath, gvk)
```
Comment (soltysh) [inline test/integration/etcd/etcd_storage_path_test.go]: Uncomment or drop it.

File: test/integration/etcd/etcd_storage_path_test.go [hunk 3]
```diff
@@ -0,0 +1,933 @@
...
+			continue
+		}
+
+		if hasTest && isEphemeral {
+			t.Errorf("duplicate test data for %s from %s.  Object has both test data and is ephemeral.", kind, pkgPath)
+			continue
+		}
+
+		if isEphemeral { // TODO it would be nice if we could remove this and infer if an object is not stored in etcd
+			// t.Logf("Skipping test for %s from %s", kind, pkgPath)
```
Comment (soltysh) [inline test/integration/etcd/etcd_storage_path_test.go]: Ucomment or drop it.

**Review by soltysh — COMMENTED** [review]
A few nits, but mostly looks good.

File: test/integration/etcd/etcd_storage_path_test.go (hunk 1 above)
Comment (enj) [inline test/integration/etcd/etcd_storage_path_test.go]: Sure, but you have to tell me how to extract protobuf from etcd without using `kapi.Codecs`.

Basically, I want to make sure no conversions, defaulting, etc happens.

Comment (soltysh) [inline test/integration/etcd/etcd_storage_path_test.go]: Right... yeah, let it be as simple as can be for now.

**Comment by soltysh** [comment]
@k8s-bot cvm gce e2e test this

**Pull request merged**